### ISSUE TITLE
Docker Use on Roadrunner

### ISSUE THREAD

**Issue by hputnam** [sr320/LabDocs#293]
I have a new user profile on Roadrunner. I downloaded and installed the docker app and have the dockerfile in my directory and get the following errors:
```hollie$ eval "$(docker-machine env default)"
Host does not exist: "default"```
```hollie$ docker build --tag="hollie/bioinformatics:v11" --file="Dockerfile.bio" .
Cannot connect to the Docker daemon. Is the docker daemon running on this host?```

Any thoughts @kubu4 @sr320

**Comment by sr320** [comment]
Did you launch the docker app once it was installed? Is the Whale at the top of the screen indicating daemon is running?

**Comment by hputnam** [comment]
When launched through GUI

![docker_issue](https://cloud.githubusercontent.com/assets/5198996/19329754/5660505a-908f-11e6-8d20-ca17068cb81d.jpg)

**Comment by hputnam** [comment]
There was another docker app hidden in the Applications. I have launched this and it seems to be working... not sure why one and not the other.

**Comment by kubu4** [comment]
What do you mean by "hidden"? I'm guessing that the original Docker install installed Docker for all accounts. However, I'm not sure why the GUI launcher you used didn't work.

Also, since you are using Docker in a new user account (instead of srlab), you might want to alter some settings. By default, I think Docker only uses 2GB of RAM on Macs. To adjust this, you'll have to exit or leave your Docker container, then launch the VirtualBox application on Roadrunner (should be in Applications), then stop the virtual machine that's running. Then, you can find the RAM settings and change them there...

This is already configured on the srlab user account to use 24GB of RAM.

**Comment by hputnam** [comment]
sorry not really hidden, just in another directory down and I didn't see it right away. 

It is working now with the older docker that shows >whale as opposed to just the whale icon.  The problem now is that the container is still building at 34.14GB. I thought it was meant to stop at ~3GB.

**Comment by kubu4** [comment]
What are you using to see the size of your image?

The Docker command to see a list of any images, and their corresponding size, is:

```docker images```

**Comment by hputnam** [comment]
I was just looking at the number increasing following the build command. When I run ```docker images```  I see:
>REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE

**Comment by kubu4** [comment]
Unfortunately, I'm not sure which number you're referring to. If you have the chance, could you post a screen shot?

I'm going to delete all images on my user account on Roadrunner and I'll use the same build command you used above and see what happens. I'll report back when I've finished.

**Comment by sr320** [comment]
In an effort to recreate (and better understand how this works) 
I tried to recreate and follow instructions on my laptop. 

Here is a screencast...
http://d.pr/i/16p4C

I think @hputnam is getting further, but any ideas what I am missing?
thanks

**Comment by kubu4** [comment]
I've encountered that issue before - not sure why it happens. Use the following command before using the eval command:

```docker-machine create -d virtualbox default```

**Comment by sr320** [comment]
for the record (see below)

@kubu4 Do not worry about trouble shooting me.- I will not be using..
But in the event we have others use it- just want to figure it out...

```
Stevens-Air:desktop sr320$ docker-machine create -d virtualbox default
Creating CA: /Users/sr320/.docker/machine/certs/ca.pem
Creating client certificate: /Users/sr320/.docker/machine/certs/cert.pem
Running pre-create checks...
Error with pre-create check: "VBoxManage not found. Make sure VirtualBox is installed and VBoxManage is in the path"
Stevens-Air:desktop sr320$ eval "$(docker-machine env default)"
Host does not exist: "default"
Stevens-Air:desktop sr320$ 
```

**Comment by kubu4** [comment]
Also, I just noticed that you launch Docker with the Docker application.

However, I always use the Docker QuickStart Terminal application that was installed when I installed Docker...

**Comment by kubu4** [comment]
Documentation for Docker install on Macs is not great. I installed using Docker Toolbox:

http://54.71.194.30:4111/mac/step_one/

That gave me the Docker QuickStart Terminal app that I use to launch Docker.

**Comment by kubu4** [comment]
It looks like Docker for Mac (which is what you and Hollie have installed) was recently released (sometime in August). This was not available when I wrote the tutorials and is not how I have been using Docker on Roadrunner. I'll have to check it out a bit and modify our documentation. In the meantime, I'll update our documentation to indicate that it is written for those using Docker Toolbox, not Docker for Mac/Windows.

**Comment by kubu4** [comment]
This difference (Docker for Mac vs. Docker Toolbox) also explains @hputnam's "incompatible CPU" message). 

The Xserve (i.e. Roadrunner, Hummingbird) CPU needs a firmware update to allow Docker for Mac to run properly (has something to do with "unlocking" CPU virtualization). I have the firmware update ready to install, but it's highly likely installing the firmware will require a restart. I'll wait to install until there's some downtime on Roadrunner, and then I'll be able to try out Docker for Mac.

**Comment by hputnam** [comment]
Now testing from srlab where it has worked before

opened Docker Quickstart Terminal and get this error:
```Error getting IP address: Something went wrong running an SSH command!
command : ip addr show
err     : exit status 255
output  :
docker is configured to use the default machine with IP 
For help getting started, check out the docs at https://docs.docker.com```

Navigate to the location of the Dockerfile.bio and type
``` eval "$(docker-machine env default)"```
Get this Error:
```Error checking TLS connection: Something went wrong running an SSH command!
command : ip addr show
err     : exit status 255
output  :```

**Comment by kubu4** [comment]
This is resolved now (I had to remove the default machine from VirtualBox). 

Have no idea why it happened - hasn't happened to me before during any of my usage.

No need to answer, but why were you using srlab account instead of your user account? Just curious.

**Comment by hputnam** [comment]
Last week Steven and I had success accessing a container from srlab, so we thought trying that again may work. It says you are using the desktop as Sam. Is it ok for me to use at srlab?

**Comment by kubu4** [comment]
Will that overtake Roadrunner? I'm currently on Roadrunner...

**Comment by hputnam** [comment]
Haha I am not sure... it might. It says "sam" is currently using the display on "roadrunner" how would you like to connect A) ask for permission to view display B) log in as yourself: "srlab"

**Comment by kubu4** [comment]
I'll leave Roadrunner.

**Comment by hputnam** [comment]
Docker build Success!! Thank you!
![docker_success](https://cloud.githubusercontent.com/assets/5198996/19372227/c978cfa6-916e-11e6-88ba-46caba7fe2da.jpg)

**Comment by hputnam** [comment]
>Sam: This is resolved now (I had to remove the default machine from VirtualBox). 

Can I remove the virtual machine also from roadrunner user hollie?
https://docs.docker.com/machine/reference/rm/

Turns out I can't access my files from srlab and can't mount owl to download from Owl to RR again.
So, I am trying again on hollie with the following error
![docker_issues_on_hollie](https://cloud.githubusercontent.com/assets/5198996/19373056/cd15b348-9175-11e6-9506-972e34ac50da.jpg)

**Comment by kubu4** [comment]
What do you mean by " can't access my files from srlab"?

You copied files to: ``` ~/Documents/Putnam/20160203_mbdseq/20160203_mbdseq```

Right? 

What is the problem you're having mounting Owl?

**Comment by hputnam** [comment]
My copied files are at ```/Users/hollie/Documents/Putnam/Data``` so I am not able to access these from another user. I tried to move them to the Shared folder and access them from there from the srlab account with working docker container, but permission was denied to access them.

So then I tried to mount owl to srlab at ```smb://owl.fish.washington.edu``` and connect with my hmp6 user name or the srlab user name and neither would mount. hmp6 won't accept my password and srlab starts to work, i select web and then it give the following error
```There was a problem connecting you to the server owl.fish.washington.edu```

I am ok not connecting if I can get docker to work on hollie.  Does it make sense to remove the default from the virtual machine also on user hollie?
https://docs.docker.com/machine/reference/rm/
